internal/provider/resource_monitor.go

```
@@ -116,13 +117,100 @@ func (r *monitorResource) Create(ctx context.Context, req resource.CreateRequest
 	}
 
 	data.Id = types.StringValue(apiResp.MonitorID)
+	
+	// Store the user's original YAML to avoid Terraform consistency check errors
+	// The normalization will be handled in Read and ModifyPlan
 	data.MonitorYaml = types.StringValue(userInputMonitorYaml)
 
 	tflog.Trace(ctx, "Created monitor resource from YAML", map[string]interface{}{"id": data.Id.ValueString()})
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
+// Helper function to safely dereference a string pointer for logging
+func derefString(s *string) string {
+	if s == nil {
+		return "<nil>"
+	}
+	return *s
+}
+
+// detectAndHandleDrift performs drift detection between state and remote YAML
+// Only updates state when there's actual semantic drift, not formatting differences
+func (r *monitorResource) detectAndHandleDrift(ctx context.Context, data *monitorResourceModel, remoteYamlBytes []byte) {
+	monitorId := data.Id.ValueString()
+	stateYaml := data.MonitorYaml.ValueString()
+
+	if stateYaml == "" || remoteYamlBytes == nil {
+		return
+	}
+
+	remoteYaml := string(remoteYamlBytes)
+
+	// Filter remote YAML to only include keys that exist in the state
+	// This prevents drift detection from triggering on server-added default fields
+	filteredRemoteYaml, err := FilterYamlKeysBasedOnTemplate(ctx, remoteYaml, stateYaml)
+	if err != nil {
+		tflog.Warn(ctx, "Failed to filter remote YAML based on state template", map[string]interface{}{
```

should we return an error in this case? i'm not sure what's TF best practice say

---

The TF best practice is that read operations should be resilient - this is why it's falling back here to using the raw yaml we got from remote. I'll improve it to error out on critical errors.